pkg/grep_test.go

```diff
@@ -0,0 +1,173 @@
+package pkg
+
+import (
+	"reflect"
+	"testing"
+)
+
+func Test_Grep(t *testing.T) {
+	testCases := []struct {
+		name        string
+		expr        string
+		in          map[string]string
+		expectedOut bool
+		isError     bool
+	}{
+		{
+			name:        "case 0",
+			expr:        "a=b,a=c",
+			in:          map[string]string{"a": "b"},
+			expectedOut: true,
+		},
+		{
+			name:        "case 1",
+			expr:        "a=b,a=c",
+			in:          map[string]string{"a": "c", "x": "y"},
+			expectedOut: true,
+		},
+		{
+			name:        "case 2",
+			expr:        "a=b,a=c",
+			in:          map[string]string{"a": "d", "x": "y"},
+			expectedOut: false,
+		},
+		{
+			name:        "case 3",
+			expr:        "a=b,a=d,x=y,x=z",
+			in:          map[string]string{"a": "d", "x": "y"},
+			expectedOut: true,
+		},
+		{
+			name:        "case 4",
+			expr:        "a!=b,x=y",
+			in:          map[string]string{"a": "d", "x": "y"},
+			expectedOut: true,
+		},
+		{
+			name:        "case 5",
+			expr:        "a!=b,x=y",
+			in:          map[string]string{"a": "b", "x": "y"},
+			expectedOut: false,
+		},
+		{
+			name:        "case 6",
+			expr:        "x!=y,x!=z",
+			in:          map[string]string{"a": "d", "x": "y"},
+			expectedOut: false,
+		},
+		{
+			name:        "case 7",
+			expr:        "x!=y,x!=z",
+			in:          map[string]string{"a": "d", "x": "z"},
+			expectedOut: false,
+		},
+		{
+			name:        "case 8",
+			expr:        "abc=xyz",
+			in:          map[string]string{"a": "d", "abc": "xyz", "x": "z"},
+			expectedOut: true,
+		},
+		{
+			name:        "case 9",
+			expr:        "abc!=xyz",
+			in:          map[string]string{"a": "d", "abc": "xyz", "x": "z"},
+			expectedOut: false,
+		},
+		{
+			name:        "case 10",
+			expr:        "abc=xyz,abc!=xyz",
+			in:          map[string]string{"a": "d", "abc": "xyz", "x": "z"},
+			expectedOut: false,
+		},
+		{
+			name:        "case 11",
+			expr:        "abc=xyz,abc!=xyz",
```

Comment:
Maybe add test cases with space in expression string (that one would normally use for readability)?

Reply:
Done.